src/Main.java

```
@@ -1,12 +1,81 @@
         task2();
         task3();
     }
-    public static void task1() { System.out.println("Задача 1"); }
-    public static void task2() { System.out.println("Задача 2"); }
+    public static void task1() {
+        System.out.println("Задача 1");
+        identifyLeap_Year(2022);
+    }
+    public static void task2() {
+        System.out.println("Задача 2");
+        byte clientOS = 0;
+        int clientDeviceYear = 2021;
+        installApp(clientOS, clientDeviceYear);
+    }
     public static void task3() {
         System.out.println("Задача 3");
+        int distance = 21;
+        deliverInDays(distance);
+    }
+    public static void identifyLeap_Year(int year) {
+        if(year % 100 != 0) {
+            if (year % 4 == 0) {
+                System.out.println(year + " год - високосный год");
+            } else {
+                System.out.println(year + " год - невисокосный год");
+            }
+        } else if(year % 400 == 0) {
+            System.out.println(year + " год - високосный год");
+        } else {
+            System.out.println(year + " год - невисокосный год");
+        }
+    }
+
+    public static void installApp(byte clientOS, int clientDeviceYear ) {
+        int currentYear = LocalDate.now().getYear();
+        //System.out.println(currentYear);
+        if(clientOS == 0) {
+            if(clientDeviceYear < currentYear) {
+                System.out.println("Установите облегченную версию приложения для iOS по ссылке");
+            } else {
+                System.out.println("Установите версию приложения для iOS по ссылке");
+            }
+        }
+        if(clientOS == 1) {
+            if(clientDeviceYear < currentYear) {
+                System.out.println("Установите облегченную версию приложения для Android по ссылке");
+            } else {
+                System.out.println("Установите версию приложения для Android по ссылке");
+            }
+        }
+        if(clientOS !=0 && clientOS !=1) {
```

вот эту проверку я двинул бы в самое начало.
Представь - ты все проверил, вызвал все методы и такой: эээ, стоп, ты вообще то неправильно вводишь.

Принцип fail fast - если что-то сразу плохо, то мы дальше даже не идем, нет смысла тратить ресурс если уже все неправильно

---

Понял. Согласен. Исправил.

---

Сделал также в третьей задаче.